src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -66,47 +64,35 @@
     @CheckForNull
     private Map<String,Object> arguments;
 
+    private Set<String> sensitiveVariables;
```

Comment:
So adding this field, the unit tests are now showing the following warning:
WARNING o.j.p.w.cps.CpsFlowExecution#maybeAutoPersistNode: Attempt to persist triggered IOException for node 7 java.lang.UnsupportedOperationException: Refusing to marshal java.util.HashMap$KeySet for security reasons; see https://jenkins.io/redirect/class-filter/
and then further down:
Caused: java.lang.RuntimeException: Failed to serialize org.jenkinsci.plugins.workflow.cps.actions.ArgumentsActionImpl#sensitiveVariables for class org.jenkinsci.plugins.workflow.cps.actions.ArgumentsActionImpl

Reply:
Hmm, well I guess you could go back to passing it around to all of the methods, convert it to an `ArrayList` instead of a set, or make the field transient. What makes the most sense depends on how the field is used and whether it would be useful for `ArgumentsAction` to have this as a persistent field or not.

Reply:
Looks like `transient` fixed the problem. Thanks! Yeah, the field should not be necessary for persistence, it's only used as an intermediary.

Reply:
Guaranteed that the Set is Hashset and not a Map.Keyset.